tests/ui/const-generics/exhaustive-value.stderr

```diff
@@ -1,8 +1,8 @@
 error[E0277]: the trait bound `(): Foo<N>` is not satisfied
-  --> $DIR/exhaustive-value.rs:262:5
+  --> $DIR/exhaustive-value.rs:262:16
    |
 LL |     <() as Foo<N>>::test()
-   |     ^^^^^^^^^^^^^^^^^^^^ the trait `Foo<N>` is not implemented for `()`
+   |                ^ the trait `Foo<N>` is not implemented for `()`
```

Comment:
Interestingly, I would have expected the span to point at `()` here, based on the output of the other tests.

Reply:
There's some subtlety in the ordering of what params to prefer over others. Right now we prefer non-self params over self params in qpaths, though arguably pointing at either of the params here is good enough.